02-array-shift/lib/array_shift.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+module.exports = exports = {};
+
+exports.insertShiftArray = (arr, item) => {
+
+  if (_isArray(arr)) {
+    let middleIndex = Math.ceil(arr.length / 2);
+    _insertMiddleItem(arr, middleIndex, item);
+
+    return arr;
+  }
+
+  throw new TypeError('insertShiftArray requires an array as its first argument');
+}
+
+function _isArray(arr) {
+  return arr instanceof Array;
+}
+
+function _insertMiddleItem(arr, middleIndex, item) {

```

Comment:
You could break this function into two separate functions if you wanted--one that shifts elements to the right, and another that inserts the middle element. Not necessary, but if you thought it'd be useful to keep the shifting right behavior, it could be good.